src/server/storage/storageAPI.js

```diff
@@ -38,16 +41,45 @@ const setup = (app: Router, storage: StorageAPI) => {
       )
         throw new Error('User email or mobile not verified!')
 
-      const user: UserRecord = defaults(body.user, {
+      const user: UserRecord = defaults(bodyUser, {
         identifier: userRecord.loggedInAs,
         createdDate: new Date().toString()
       })
+
       //mautic contact should already exists since it is first created during the email verification we update it here
       const mauticRecord = process.env.NODE_ENV === 'development' ? {} : await Mautic.createContact(user).catch(e => {})
       const mauticId = get(mauticRecord, 'contact.fields.all.id', -1)
       logger.debug('User mautic record', { mauticId, mauticRecord })
+
+      const updateUserObj = {
+        ...user,
+        mauticId
+      }
+
+      if (!user.w3Token) {
+        try {
+          const secureHash = md5(`${user.email}${conf.secure_key}`)
+          const web3Response = await fetch(`${conf.web3SiteUrl}/api/wl/user`, {
+            method: 'PUT',
+            body: {
+              secure_hash: secureHash,
+              email: user.email,
+              full_name: user.fullName,
+              wallet_address: walletAddress
+            }
+          }).then(res => res.json())
+
+          if (web3Response.wallet_token) {
+            updateUserObj.w3Token = web3Response.wallet_token
+          }
+        } catch (e) {
+          log.debug('Get Web3 Login Response Failed', e)
+        }
+      }
+
+      storage.updateUser(updateUserObj)
+
       //topwallet of user after registration
-      storage.updateUser({ ...user, mauticId })
       let ok = await AdminWallet.topWallet(userRecord.gdAddress, null, true)
```

Comment:
I think we need to return the W3 token to the user, to be saved on his profile (encrypted)
So on the UI we can create the link to w3.gooddollar.org with the login token